rust/signed_doc/tests/integration_test.rs

```diff
@@ -0,0 +1,253 @@
+//! Integration test for the `catalyst_signed_doc` crate.
```

Comment:
better to split each test case under their own file.
So group them something like this `decoding.rs`, `signature.rs`,`proposal.rs`, `comment.rs`